### PR TITLE
[IMP] sale_commission: base amount for computation selectable

### PR DESCRIPTION
**Pull request by oihane**
You can now select if commission is calculated from subtotal
or from margin

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Include the new possibilities on the README.rst file.

File: sale_commission/models/account_invoice.py [hunk 1]
```diff
@@ -124,17 +124,23 @@ class AccountInvoiceLineAgent(models.Model):
     def onchange_agent(self):
         self.commission = self.agent.commission
 
-    @api.one
-    @api.depends('commission.commission_type', 'invoice_line.price_subtotal')
+    @api.depends('commission.commission_type', 'invoice_line.price_subtotal',
+                 'commission.amount_base_type')
     def _get_amount(self):
```
Comment (pedrobaeza) [inline sale_commission/models/account_invoice.py]: rename it to `_compute_amount` according with last standards.

File: sale_commission/models/sale_commission.py [hunk 2]
```diff
@@ -40,6 +40,10 @@ class SaleCommission(models.Model):
         [('open', 'Invoice Based'),
          ('paid', 'Payment Based')], string='Invoice Status',
         required=True, default='open')
+    amount_base_type = fields.Selection(
+        selection=[('subtotal', 'Subtotal'),
```
Comment (pedrobaeza) [inline sale_commission/models/sale_commission.py]: Call it "Gross amount"

File: sale_commission/models/sale_commission.py [hunk 3]
```diff
@@ -40,6 +40,10 @@ class SaleCommission(models.Model):
...
+                   ('margin', 'Margin')],
```
Comment (pedrobaeza) [inline sale_commission/models/sale_commission.py]: Call it "Net amount", and put a help explaining a bit more what is each thing.

File: sale_commission/models/sale_order.py [hunk 4]
```diff
@@ -88,19 +88,24 @@ class SaleOrderLineAgent(models.Model):
...
-    @api.one
     @api.onchange('agent')
     def onchange_agent(self):
         self.commission = self.agent.commission
 
-    @api.one
-    @api.depends('commission.commission_type', 'sale_line.price_subtotal')
+    @api.depends('commission.commission_type', 'sale_line.price_subtotal',
+                 'commission.amount_base_type')
     def _get_amount(self):
```
Comment (pedrobaeza) [inline sale_commission/models/sale_order.py]: Call it also `_compute_amount`

File: sale_commission/README.rst [hunk 5]
```diff
@@ -7,6 +12,9 @@ them in customers and sales orders.
 You can then make the settlements of these commissions, and generate the
 corresponding supplier invoices to pay their commissions fees.
 
+You can define which base amount are going to take into account, net amount
```
Comment (pedrobaeza) [inline sale_commission/README.rst]: s/are going to take into account,/is going to be taken into account:

**Comment by pedrobaeza** [comment]
Thanks for the complete test suite. Definitively :+1: , but just 2 minor things:
- You can replace `(c)` by `©`.
- People from Akretion are not credited, and they should, because they developed this PR: https://github.com/OCA/commission/pull/40. @rvalyi, who we should credit?

**Comment by pedrobaeza** [comment]
Ah, and please squash the commits.

**Comment by oihane** [comment]
Something went really bad when making the squash...
A part of the code has disappeared... Please @pedrobaeza set again **work in progress**

**Comment by pedrobaeza** [comment]
Again :+1:

**Comment by Daniel-CA** [comment]
:+1:

File: sale_commission/models/account_invoice.py [hunk 6]
```diff
@@ -112,39 +96,44 @@ class AccountInvoiceLineAgent(models.Model):
...
-            subtotal = self.invoice_line.price_subtotal
-            if self.commission.commission_type == 'fixed':
-                self.amount = subtotal * (self.commission.fix_qty / 100.0)
-            else:
-                self.amount = self.commission.calculate_section(subtotal)
-
-    @api.one
+    @api.depends('commission.commission_type', 'invoice_line.price_subtotal',
+                 'commission.amount_base_type')
+    def _compute_amount(self):
```
Comment (eLBati) [inline sale_commission/models/account_invoice.py:112]: In general, I would not change method names of stable modules, as there could be other modules using/extending them, that would suddenly stop working

Comment (pedrobaeza) [inline sale_commission/models/account_invoice.py:112]: This change is for following guidelines, but I agree that it can break something.

**Pull request merged**